binstar_client/utils/__init__.py

```diff
@@ -247,8 +245,8 @@ def recursive_update(d, u):
             d[k] = u[k]
     return d
 
-DEFAULT_URL = 'https://api.anaconda.org'
```

Comment:
That happens at startup to set the default, but when this line executes https://github.com/Anaconda-Server/anaconda-client/blob/api-on-premises/binstar_client/utils/__init__.py#L161, what's in the config file will take effect regardless of the default. If you want to make the env variable take precedence, you will have to change that line.